File: content/examples/other/leetcode/problems/maximum-subarray/example-1/Solution.java
public class Solution {
  public static int maxSubArray(int[] nums) {
    int best = Integer.MIN_VALUE;
    for (int i = 0; i < nums.length; i++) {
      int sum = 0;
      for (int j = i; j < nums.length; j++) {
        sum += nums[j];
        best = Math.max(best, sum);
      }
    }
    return best;
  }

  public static void main(String[] args) {
    System.out.println(maxSubArray(new int[]{-2,1,-3,4,-1,2,1,-5,4}));
  }
}

File: content/examples/other/leetcode/problems/maximum-subarray/example-1/solution.js
function maxSubArray(nums) {
  let best = -Infinity;
  for (let i=0;i<nums.length;i+=1) {
    let sum = 0;
    for (let j=i;j<nums.length;j+=1) {
      sum += nums[j];
      best = Math.max(best, sum);
    }
  }
  return best;
}

if (require.main===module) {
  console.log(maxSubArray([-2,1,-3,4,-1,2,1,-5,4]));
}

module.exports = { maxSubArray };
